custom/icds_reports/views.py

```diff
@@ -2164,3 +2170,43 @@ def get(self, request, *args, **kwargs):
         except Exception:
             response = dict(isSuccess=False, message='Unknown Error occured')
             return JsonResponse(response, status=500)
+
+
+@location_safe
+class BaseGovernanceAPIView(View):
+    def get_settings(self, request, *args, **kwargs):
+        step = kwargs.get('step')
+        now = datetime.utcnow()
+        month = int(request.GET.get('month', now.month))
+        year = int(request.GET.get('year', now.year))
+
+        if (now.day == 1 or now.day == 2) and now.month == month and now.year == year:
+            prev_month = now - relativedelta(months=1)
+            month = prev_month.month
+            year = prev_month.year
+
+        start = request.GET.get('start', 0)
+
+        return step, start, month, year
+
+
+@method_decorator(DASHBOARD_CHECKS, name='dispatch')
+class GovernanceAPIView(BaseGovernanceAPIView):
+
+    def get(self, request, *args, **kwargs):
+        step, start, month, year = self.get_settings(request, *args, **kwargs)
+
+        if step == 'home_visit':
+            length = GOVERNANCE_API_HOME_VISIT_RECORDS_PAGINATION
+            location_filters = {'aggregation_level': 5}
```

Comment:
Well, technically this is not a location filter. You could just call it something like query_filter.

Reply:
done